vigor/ajax.py

```diff
@@ -0,0 +1,61 @@
+import json
+from django.conf import settings
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.http import HttpResponse, HttpResponseRedirect, HttpResponseForbidden
+from django.shortcuts import get_object_or_404
+from django.views.decorators.csrf import csrf_exempt
+from .models import SystemMessage, UserPrefs
+
+def response_mimetype(req):
+    if "application/json" in req.META['HTTP_ACCEPT']:
+        return "application/json"
+    else:
+        return "text/plain"
+
+class JSONResponse(HttpResponse):
+    """JSON response class."""
+
+    def __init__(self, obj=None, json_opts=None, mimetype="application/json", *args, **kwargs):
+        if obj is None:
+            obj = {}
+        if json_opts is None:
+            json_opts = {}
+        content = json.dumps(obj, **json_opts)
+        super(JSONResponse, self).__init__(content, mimetype, *args, **kwargs)
+
+def sign_out(req):
+    logout(req)
+    # next = req.GET.get('next', None)
+    # if next:
+    #     return HttpResponseRedirect(next)
+    # else:
+    return HttpResponseRedirect(reverse('login'))
+
+@csrf_exempt
+def register(req):
+    if settings.REGISTER:
+        if req.method == "POST":
+            username = req.POST['username']
+            password = req.POST['password']
+            email = req.POST['email']
+
+            user = User.objects.create_user(username, email, password)
+            UserPrefs.objects.create(user = user)
+            user2 = authenticate(username = username, password = password)
+            login(req, user2)
```

Comment:
That's actually a really great point. I wanted to look into this, but since I was on such a short time frame to get this done, I just wanted to get this done as quickly as I could. I will render this in the next version.